### PR TITLE
New SCIP interface

### PR DESCRIPTION
**Pull request by SteveDiamond**
## Description
This interface works with the latest pyscipopt and SCIP. It does not provide dual variables for SOCPs, only for LPs.
Issue link (if applicable): https://github.com/cvxpy/cvxpy/issues/1889

## Type of change
- [x] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: continuous_integration/install_dependencies.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ fi
 
 # SCIP only works with scipy >= 1.5 due to dependency conflicts when installing on Linux/macOS
 if [[ "$PYTHON_VERSION" == "3.9" ]] || [[ "$RUNNER_OS" == "Windows" ]]; then
-  conda install pyscipopt"<4.0"  # TODO: update interface https://github.com/cvxpy/cvxpy/pull/1628
+  conda install pyscipopt  # TODO: update interface https://github.com/cvxpy/cvxpy/pull/1628
```
Comment (phschiele) [inline continuous_integration/install_dependencies.sh:34]:
```suggestion
  conda install pyscipopt
```
This comment is resolved by this PR.

**Review by phschiele — APPROVED** [review]
- Tested that interface works with `pyscipopt 3.5` (`scip 7`) as well as `pyscipopt 4.2` (`scip 8`).
- Had to change to `places=2` for `test_scip_socp_1` when using `4.2`. Perhaps we should loosen the tolerance to ensure its passing for all users (can't make the change here directly short of creating a separate PR pointing to this one). 
- docs are building
- coverage is looking good (97%), but not shown here because we can't install `pysciopt` on the mac os/3.8, which is running the coverage report.

File: cvxpy/reductions/solvers/conic_solvers/scip_conif.py [hunk 2]
```diff
@@ -265,16 +260,20 @@ def _set_params(
...
-        model.setPresolve(SCIP_PARAMSETTING.OFF)
-        model.setHeuristics(SCIP_PARAMSETTING.OFF)
-        model.disablePropagation()
+        is_mip = data[s.BOOL_IDX] or data[s.INT_IDX]
+        has_soc_constr = len(dims[s.SOC_DIM]) > 1
+        if not (is_mip or has_soc_constr):
+            # These settings are needed  to allow the dual to be calculated
+            model.setPresolve(SCIP_PARAMSETTING.OFF)
+            model.setHeuristics(SCIP_PARAMSETTING.OFF)
+            model.disablePropagation()
```
Comment (phschiele) [inline cvxpy/reductions/solvers/conic_solvers/scip_conif.py:276]: There could be an edge case where these settings are overwritten by the user later.

Comment (rileyjmurray) [reply cvxpy/reductions/solvers/conic_solvers/scip_conif.py:276]: @phschiele could we raise a warning if the user tries to change either of these values?

Comment (SteveDiamond) [reply cvxpy/reductions/solvers/conic_solvers/scip_conif.py:276]: Isn't the model object created fresh each time we solve the problem though? Or are these settings globals?

Comment (phschiele) [reply cvxpy/reductions/solvers/conic_solvers/scip_conif.py:276]: I was referring to the case where the user specifies these options as kwargs which are parsed a few lines below. Raising a warning as Riley suggested would be good imo.

**Pull request merged**